### PR TITLE
Billing and Reporting Requirements

### PR DESCRIPTION
**Pull request by tobru**
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/references/functional-requirements/billing.adoc [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+|===
+|Description
+|Quantity
+|Unit of Measure
+|Unit Price
+|Discount
+|Total (Rounded)
+
+6+e|*Compute*
+
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/functional-requirements/billing.adoc:36]: I looked at the preview and can't help but wonder if a sorting by namespace makes more sense than sort by product category.

The current example lists all compute resources by zone and namespace.
Then it lists all storage resources by zone and namespace.
Then maybe LoadBalancer, or other categories, by zone and namespace and so on.

This makes it hard to follow what exactly a namespace is costing me, as I'd have to search and sum up resources from all over the place.

What if instead the line items are combined and summarized by namespace instead?

For example (just prosa ASCII art):
```
                                  Quantity | Unit  | ...
========================================================
zone: cloudscale.ch LPG 2
namespace: my-app-prod

Compute
--------------------------------------------------------
Usage (Min / Max / Avg): 
128 MiB / 1034 MiB / 585.66 MiB  |  7028   |  MiB  |  ...
--------------------------------------------------------

Persistent Storage (PVC)
--------------------------------------------------------
Usage (Avg): 
10 GiB                           |  100    |  GiB  |  ...
--------------------------------------------------------

Total for namespace
--------------------------------------------------------
                                                    0.02

========================================================
zone: Exoscale GVA 1
namespace: my-app-test

Compute
--------------------------------------------------------
Usage (Min / Max / Avg): 
...
```

Ofc, not all namespaces have storage. and some may have storage without compute. In those cases the listing for a category for each namespace just disappears. The "Total for namespace" then basically gives summarizes the cost for a namespace.

A decision-maker could then better decide where to optimize cost for applications. e.g it allows to say "this test app costs us x amount per month. Do we need it?"

Comment (corvus-ch) [reply docs/modules/ROOT/pages/references/functional-requirements/billing.adoc:36]: I second that idea.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
I would like to see the suggestion of @ccremer to be applied.

From my understanding of Odood, that could be doable. It would require the adapter to create and use catecories based on zone and namespaces.

**Comment by tobru** [comment]
It absolutely makes much more sense to group by Zone/NS. I updated the example accordingly. Please have a look again @ccremer @corvus-ch 

Do we also need to talk about VAT here or is this something which has to be tackled by the ERP / Billing System?

File: docs/modules/ROOT/pages/references/functional-requirements/billing.adoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+Account ID:: Hardcoded to 49.
+Currency ID:: Hardcoded to 6 (CHF).
+Invoice date:: Date of the invoice.
+Journal ID:: Hardcoded to 1.
+Name:: Name of the invoice.
+Payment term:: Hardcoded to 3.
+State:: Draft.
+User ID:: The ID of the user who created the invoice.
+
+=== Per Line Item
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/references/functional-requirements/billing.adoc:101]: I remember we talked about not to use the Odoo internal price table but use our own. For this I am missing a field to enter the chared amount.

Comment (tobru) [reply docs/modules/ROOT/pages/references/functional-requirements/billing.adoc:101]: Yes, that was missing. Added.

**Review by ccremer — APPROVED** [review]
Example invoice LGTM, though I can't judge if the fields are correct in terms of Odoo

**Review by corvus-ch — APPROVED** [review]
